### PR TITLE
fix: Updated Input stories to show appearance and disabled

### LINKED ISSUE #32174: [Bug]: v9 Input stories missing shadow appearances
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting an Accessibility issue?

None

### Reproduction

http://localhost:3000/?path=/docs/components-input--default#appearance

## Actual Behavior
filled lighter and darker shadow appearances are in storybook documentation

## Expected Behavior
stories are missing

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### LINKED ISSUE #32175: [Bug]: v9 Input stories missing appearances for disabled and toggle ability
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting an Accessibility issue?

None

### Reproduction

http://localhost:3000/?path=/docs/components-input--default#disabled

## Actual Behavior
The disabled story example only includes the default appearance, so a representation of the disabled state for each appearance is missing.  Because of transition CSS, there needs to be a way to toggle disabled state to see the change.

## Expected Behavior
Each appearance is an example for disabled
There is a toggle for disabled in the disabled examples

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Previous Behavior
- appearance stories missing shadow variants
- disabled stories missing appearance variants
- no way to toggle disabled to see transition css

## New Behavior
- variants added to stories
- can toggle disabled

## Related Issue(s)
- Fixes #32174
- Fixes #32175

## Screenshots (after)

image

image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-input/stories/src/Input/InputAppearance.stories.tsx [hunk 1]
```diff
@@ -28,32 +28,43 @@ const useStyles = makeStyles({
...
+        <Input appearance="underline" id={inputIds.underline} />
       </div>
-
       <div className={mergeClasses(styles.field, styles.filledLighter)}>
-        <Label htmlFor={filledLighterId}>Filled lighter appearance</Label>
-        <Input appearance="filled-lighter" id={filledLighterId} />
+        <Label htmlFor={inputIds.filledLighter}>Filled lighter appearance</Label>
+        <Input appearance="filled-lighter" id={inputIds.filledLighter} />
+      </div>
+      <div className={styles.field}>
```
Comment (spmonahan) [inline packages/react-components/react-input/stories/src/Input/InputAppearance.stories.tsx:57]: Does this also need `styles.filledLighter`?

```suggestion
      <div className={mergeClasses(styles.field, styles.filledLighter)}>
```

File: packages/react-components/react-input/stories/src/Input/InputAppearance.stories.tsx [hunk 2]
```diff
@@ -28,32 +28,43 @@ const useStyles = makeStyles({
...
+        <Input appearance="filled-lighter-shadow" id={inputIds.filledLighterShadow} />
       </div>
-
       <div className={mergeClasses(styles.field, styles.filledDarker)}>
-        <Label htmlFor={filledDarkerId}>Filled darker appearance</Label>
-        <Input appearance="filled-darker" id={filledDarkerId} />
+        <Label htmlFor={inputIds.filledDarker}>Filled darker appearance</Label>
+        <Input appearance="filled-darker" id={inputIds.filledDarker} />
+      </div>
+      <div className={styles.field}>
```
Comment (spmonahan) [inline packages/react-components/react-input/stories/src/Input/InputAppearance.stories.tsx:65]: Filled darker style?

```suggestion
      <div className={mergeClasses(styles.field, styles.filledDarker)}>
```

**Review by spmonahan — APPROVED** [review]

File: packages/react-components/react-input/stories/src/Input/InputDisabled.stories.tsx [hunk 3]
```diff
@@ -1,26 +1,101 @@
...
+    default: useId(idPrefix),
+    underline: useId(idPrefix),
+    filledLighter: useId(idPrefix),
+    filledLighterShadow: useId(idPrefix),
+    filledDarker: useId(idPrefix),
+    filledDarkerShadow: useId(idPrefix),
+  };
   const styles = useStyles();
 
+  let [disabled, setDisabled] = React.useState(true);
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-input/stories/src/Input/InputDisabled.stories.tsx:45]:
```suggestion
  const [disabled, setDisabled] = React.useState(true);
```

**Review by smhigley — CHANGES_REQUESTED** [review]
The shadow appearances would need to be shown over a dark background, since on a light background they fail WCAG's non-text contrast criterion (the boundary must meet 3:1 contrast).

I think this is what was brought up in chat, but I think those might also be deprecated/not different from filled darker & lighter?

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Avatar-mount_base.html">641</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Avatar-mount.html">642</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Button-mount_base.html">298</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Button-mount.html">290</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Field-mount_base.html">1106</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Field-mount.html">1173</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProvider-mount_base.html">729</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProvider-mount.html">743</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-mount_base.html">82</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-mount.html">87</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">34</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">39</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">85</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/MakeStyles-mount_base.html">865</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/MakeStyles-mount.html">889</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Persona-mount_base.html">1763</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/Persona-mount.html">1743</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/SpinButton-mount_base.html">1371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/SpinButton-mount.html">1397</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/SwatchPicker-mount_base.html">1644</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32176/perf-test-react-components/SwatchPicker-mount.html">1642</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-input-a0d70a8b-d328-4827-81af-cebe154a4b80.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-input-a0d70a8b-d328-4827-81af-cebe154a4b80.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32176&commitId=7274eb9f71ae545af8849966999f2c8d190edaba&env=prod&clientType=FLUENTUI)


  <details><summary><b>SwatchPicker Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|SwatchPicker Converged.Spacing.default.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by smhigley — APPROVED** [review]

File: change/@fluentui-react-input-a0d70a8b-d328-4827-81af-cebe154a4b80.json (hunk 4 above)
Comment (GeoffCoxMSFT) [reply change/@fluentui-react-input-a0d70a8b-d328-4827-81af-cebe154a4b80.json:1]: Spurious report

**Pull request merged**